File: backend/app/core/log_levels.py
"""日志级别常量定义

统一管理系统中使用的日志级别，避免重复定义。
"""

from enum import Enum
from typing import List


class LogLevel(str, Enum):
    """日志级别枚举"""
    DEBUG = "DEBUG"
    INFO = "INFO"
    WARNING = "WARNING"
    ERROR = "ERROR"
    CRITICAL = "CRITICAL"


# 日志级别列表
LOG_LEVELS: List[str] = [level.value for level in LogLevel]

# 有效的日志级别（用于验证）
VALID_LOG_LEVELS = LOG_LEVELS

# 允许的日志级别（用于配置验证）
ALLOWED_LOG_LEVELS = LOG_LEVELS

File: frontend/src/constants/logLevels.ts
// 日志级别常量定义
export const LOG_LEVELS = {
  DEBUG: 'DEBUG',
  INFO: 'INFO',
  WARNING: 'WARNING',
  ERROR: 'ERROR',
  CRITICAL: 'CRITICAL'
} as const;

// 日志级别类型（大写）
export type LogLevel = typeof LOG_LEVELS[keyof typeof LOG_LEVELS];

// 日志级别类型（小写，兼容system.ts）
export type LogLevelLowercase = 'debug' | 'info' | 'warning' | 'error' | 'critical';

// 日志级别选项（用于下拉选择）
export const logLevelOptions = [
  { label: 'DEBUG', value: 'DEBUG' },
  { label: 'INFO', value: 'INFO' },
  { label: 'WARNING', value: 'WARNING' },
  { label: 'ERROR', value: 'ERROR' },
  { label: 'CRITICAL', value: 'CRITICAL' }
];

// 日志级别颜色映射
export const logLevelColors = {
  DEBUG: '#909399',
  INFO: '#409EFF',
  WARNING: '#E6A23C',
  ERROR: '#F56C6C',
  CRITICAL: '#F56C6C'
} as const;